include/libopencm3/stm32/h7/rcc.h

```diff
@@ -158,8 +180,133 @@ LGPL License Terms @ref lgpl_license
+#define RCC_D2CFGR_RSVD_BITMASK     (0xfffff88f)
+/**@}*/
 
-/** @addtogroup rcc_bdcr_values RCC_BDCR_VALUES
+/** @defgroup rcc_d3cfgr_values RCC_D3CFGR Values
+ * @ingroup rcc_registers
+ * @{*/
+#define RCC_D3CFGR_D3PPRE_BYP       0x0
+#define RCC_D3CFGR_D3PPRE_DIV2      0x4
+#define RCC_D3CFGR_D3PPRE_DIV4      0x5
+#define RCC_D3CFGR_D3PPRE_DIV8      0x6
+#define RCC_D3CFGR_D3PPRE_DIV16     0x7
+#define RCC_D3CFGR_D3PPRE_SHIFT     4
+#define RCC_D3CFGR_RSVD_BITMASK     (0xffffff8f)
+/**@}*/
+
+/** @defgroup rcc_pllckselr_values RCC_PLLCKSELR Values
+ * @ingroup rcc_registers
+ * @{*/
+#define RCC_PLLCKSELR_PLLSRC_HSI    0x0
+#define RCC_PLLCKSELR_PLLSRC_CSI    0x1
+#define RCC_PLLCKSELR_PLLSRC_HSE    0x2
+#define RCC_PLLCKSELR_PLLSRC_NONE   0x3
+#define RCC_PLLCKSELR_DIVM_DIS      0
+#define RCC_PLLCKSELR_DIVM_BYP      1
+#define RCC_PLLCKSELR_DIVM_MASK     0x3f
+
+#define RCC_PLLCKSELR_DIVM3_SHIFT   20
+#define RCC_PLLCKSELR_DIVM2_SHIFT   12
+#define RCC_PLLCKSELR_DIVM1_SHIFT   4
+
+#define RCC_PLLCKSELR_DIVM3(n)      ((n) << RCC_PLLCKSELR_DIVM3_SHIFT)
+#define RCC_PLLCKSELR_DIVM2(n)      ((n) << RCC_PLLCKSELR_DIVM2_SHIFT)
+#define RCC_PLLCKSELR_DIVM1(n)      ((n) << RCC_PLLCKSELR_DIVM1_SHIFT)
+/**@}*/
+
+/** @defgroup rcc_pllcfgr_values RCC_PLLCFGR Values
+ * @ingroup rcc_registers
+ * @{*/
+#define RCC_PLLCFGR_PLLRGE_1_2MHZ   0
+#define RCC_PLLCFGR_PLLRGE_2_4MHZ   1
+#define RCC_PLLCFGR_PLLRGE_4_8MHZ   2
+#define RCC_PLLCFGR_PLLRGE_8_16MHZ  3
+
+#define RCC_PLLCFGR_DIVR3EN         BIT24
+#define RCC_PLLCFGR_DIVQ3EN         BIT23
+#define RCC_PLLCFGR_DIVP3EN         BIT22
+#define RCC_PLLCFGR_DIVR2EN         BIT21
+#define RCC_PLLCFGR_DIVQ2EN         BIT20
+#define RCC_PLLCFGR_DIVP2EN         BIT19
+#define RCC_PLLCFGR_DIVR1EN         BIT18
+#define RCC_PLLCFGR_DIVQ1EN         BIT17
+#define RCC_PLLCFGR_DIVP1EN         BIT16
+#define RCC_PLLCFGR_PLL3RGE_SHIFT   10
+#define RCC_PLLCFGR_PLL3VCO_WIDE    0       /* 192 - 836MHz base output. */
+#define RCC_PLLCFGR_PLL3VCO_MED     BIT9    /* 150 - 420MHz base output. */
+#define RCC_PLLCFGR_PLL3FRACEN      BIT8
+#define RCC_PLLCFGR_PLL2RGE_SHIFT   6
+#define RCC_PLLCFGR_PLL2VCO_WIDE    0       /* 192 - 836MHz base output. */
+#define RCC_PLLCFGR_PLL2VCO_MED     BIT5    /* 150 - 420MHz base output. */
+#define RCC_PLLCFGR_PLL2FRACEN      BIT4
+#define RCC_PLLCFGR_PLL1RGE_SHIFT   2
+#define RCC_PLLCFGR_PLL1VCO_WIDE    0       /* 192 - 836MHz base output. */
+#define RCC_PLLCFGR_PLL1VCO_MED     BIT1    /* 150 - 420MHz base output. */
+#define RCC_PLLCFGR_PLL1FRACEN      BIT0
+#define RCC_PLLCFGR_RSVD_BITMASK    (0xfe00f000)
+/**@}*/
+
+/** @defgroup rcc_plldivr_values RCC_PLLnDIVR Values
+ * @ingroup rcc_registers
+ * @{*/
+#define RCC_PLLNDIVR_DIVR_SHIFT     24
+#define RCC_PLLNDIVR_DIVQ_SHIFT     16
+#define RCC_PLLNDIVR_DIVP_SHIFT     9
+#define RCC_PLLNDIVR_DIVN_SHIFT     0
+
+/* Need to preserve reserved bits, so give easy mask shortcut. */
+#define RCC_PLLNDIVR_RSVD_BITMASK   (0x80800100)
```

Comment:
Here you've even artificually included one of the "illegal values" from a field, that's not actually reserved, making this even harder to track in the future.